packages/material-ui/src/Rating/Rating.js

```diff
@@ -340,15 +332,14 @@ const Rating = React.forwardRef(function Rating(props, ref) {
 
     return (
       <React.Fragment key={state.value}>
-        <label className={classes.label} htmlFor={id} {...labelProps}>
+        <label className={classes.label} {...labelProps}>
```

Comment:
I'm a little unsure how this will affect a11y. With the label properly pointing to the input, it was triggering the `onClick` handler twice, causing it to unset every time.

Reply:
I might be able to add the `htmlFor` back in and make the click handler bail early if hover is `-1`

Reply:
OK, yeah that worked, though still not sure how the `readOnly` flag will affect a11y since it's not really readOnly. I'd imagine not having this properly hooked up and using the `onChange` handler would make it difficult or impossible to activate from a screen reader.